### PR TITLE
libcs50: init at 11.0.2

### PR DESCRIPTION
**Pull request by t-spr**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libcs50/default.nix
Comment (drupol) [inline pkgs/development/libraries/libcs50/default.nix:3]: I think we should use the `finalAttrs` pattern here. Here's a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

**Review by drupol — CHANGES_REQUESTED** [review]
I added a comment in your proposed code.

You also need to update the commit message: `libcs50: init at 11.0.2` and this PR title as well.

File: pkgs/development/libraries/libcs50/default.nix
Comment (t-spr) [reply pkgs/development/libraries/libcs50/default.nix]: Thanks for the feedback and documentation links, I think I've got it now.

File: pkgs/development/libraries/libcs50/default.nix
Comment (drupol) [inline pkgs/development/libraries/libcs50/default.nix:15]: Missing hooks.

Check this derivation as example on how to add them: https://github.com/NixOS/nixpkgs/pull/234428

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix
Comment (t-spr) [reply pkgs/top-level/all-packages.nix]: Yes, I agree everything should be in alphabetical order. I moved it to a spot that might be more appropriate but it looks like most people just been adding packages to the end of the file.

File: pkgs/development/libraries/libcs50/default.nix
Comment (drupol) [inline pkgs/development/libraries/libcs50/default.nix:10]:
```suggestion
    rev = "v${finalAttrs.version}";
```

File: pkgs/development/libraries/libcs50/default.nix
Comment (t-spr) [reply pkgs/development/libraries/libcs50/default.nix:24]: I don't think the project has a changelog, at least I couldn't find one

**Review by drupol — CHANGES_REQUESTED** [review]
I just spotted a minor thing to fix.

After that, I'll leave the final decision to another maintainer because I'm a bit unsure on the `installPhase`.

File: pkgs/top-level/all-packages.nix
Comment (drupol) [reply pkgs/top-level/all-packages.nix]: Yeah it's a bit messy sometimes, but let's try to improve reduce the entropy of the file when we can :)

File: pkgs/development/libraries/libcs50/default.nix
Comment (drupol) [reply pkgs/development/libraries/libcs50/default.nix:24]: Fair enough.

**Review by drupol — APPROVED** [review]

**Pull request merged**
